aiosmtpd/smtp.py

```diff
@@ -26,6 +27,23 @@
 EMPTYBYTES = b''
 
 
+class Session:
+    peer = None
+    ssl = None
+    host_name = None
+    extended_smtp = False
+
+
+class Envelope:
+    mail_from = None
+    mail_options = None
+    received_data = None
+
+    def __init__(self):
+        self.rcpt_tos = []
+        self.rcpt_options = []
```

Comment:
My logic to where place parameters is based on RSET and post data state. If one of them drops param value - then it is a part of envelope. Otherwice - part of session.
Maybe thats a confusing name as meaning of this variable little bit changed. Renamed it to just "data", is it contains email content itself, while "received data" is a local variable of DATA command. May be better to call it "envelope.body" or "envelope.content", not sure.